scripts/kconfig/kconfigfunctions.py

```diff
@@ -349,6 +349,22 @@ def dt_compat_enabled(kconf, _, compat):
     return "n"
 
 
+def dt_nodelabel_has_compat(kconf, _, label, compat):
+    """
+    This function takes a 'label' and returns "y" if an "enabled" node with
+    such label can be found in the EDT and that node is compatible with the
+    provided 'compat', otherwise it returns "n".
+    """
+    if doc_mode or edt is None:
+        return "n"
+
+    for node in edt.nodes:
```

Comment:
I believe it would be faster to look up the enabled nodes with this compatible using `edt.compat2enabled[compat]`. This will avoid looping over all the nodes in the devicetree each time this function is called.

Then each node can be checked as `if label in node.labels: return "y"`.

Reply:
Thanks for the hint. Improved as suggested.